packages/@aws-cdk/mixins-preview/lib/util/property-mixins.ts

```diff
@@ -0,0 +1,175 @@
+import { AssumptionError } from 'aws-cdk-lib/core';
+
+/**
+ * Deep merge utility for nested objects
+ * @param target The target object to merge into
+ * @param source The source object to merge from
+ * @param mergeOnly The explicit list of property keys to copy from source
+ */
+export function deepMerge(target: any, source: any, mergeOnly: string[]): any {
+  for (const key of mergeOnly) {
+    if (key === '__proto__' || key === 'constructor' || key === 'prototype') {
+      continue;
+    }
+
+    if (!(key in source)) {
+      continue;
+    }
+
+    const sourceValue = source[key];
+    const targetValue = target[key];
+
+    if (typeof sourceValue === 'object' && sourceValue != null && !Array.isArray(sourceValue) &&
+        typeof targetValue === 'object' && targetValue != null && !Array.isArray(targetValue)) {
+      target[key] = deepMergeCopy({}, targetValue, sourceValue);
+    } else {
+      target[key] = sourceValue;
+    }
+  }
+
+  return target;
+}
+
+/**
+ * Object keys that deepMerge should not consider. Currently these include
+ * CloudFormation intrinsics
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html
+ */
+
+const MERGE_EXCLUDE_KEYS: string[] = [
+  'Ref',
+  'Fn::Base64',
+  'Fn::Cidr',
+  'Fn::FindInMap',
+  'Fn::GetAtt',
+  'Fn::GetAZs',
+  'Fn::ImportValue',
+  'Fn::Join',
+  'Fn::Select',
+  'Fn::Split',
+  'Fn::Sub',
+  'Fn::Transform',
+  'Fn::And',
+  'Fn::Equals',
+  'Fn::If',
+  'Fn::Not',
+  'Fn::Or',
+];
+
+/**
+ * Merges `source` into `target`, overriding any existing values.
+ * `null`s will cause a value to be deleted.
+ */
+function deepMergeCopy(target: any, ...sources: any[]) {
+  for (const source of sources) {
+    if (typeof(source) !== 'object' || typeof(target) !== 'object') {
+      throw new AssumptionError(`Invalid usage. Both source (${JSON.stringify(source)}) and target (${JSON.stringify(target)}) must be objects`);
+    }
+
+    for (const key of Object.keys(source)) {
+      if (key === '__proto__' || key === 'constructor') {
```

Comment:
👏

Reply:
added check to existing deepMerges in the library